mne_bids/utils.py

```diff
@@ -621,3 +623,79 @@ def _find_matching_sidecar(bids_fname, bids_root, suffix, allow_fail=False):
         return None
     else:
         raise RuntimeError(msg)
+
+
+def _update_sidecar(sidecar_fname, key, val):
+    """Update a sidecar JSON file with a given key/value pair.
+
+    Parameters
+    ----------
+    sidecar_fname : str
+        Full name of the data file
+    key : str
+        The key in the sidecar JSON file. E.g. "PowerLineFrequency"
+    val : str
+        The corresponding value to change to in the sidecar JSON file.
+    """
+    with open(sidecar_fname, "r") as fin:
+        sidecar_json = json.load(fin)
+    sidecar_json[key] = val
+    with open(sidecar_fname, "w") as fout:
+        json.dump(sidecar_json, fout)
+
+
+def _estimate_line_noise(raw, verbose=False):
+    """Estimate power line noise from a given BaseRaw.
+
+    Parameters
+    ----------
+    raw : mne.io.BaseRaw
+
+    Returns
+    -------
+    powerlinefrequency : int
+        Either 50, or 60 Hz depending if European,
+        or USA data recording.
+    """
+    sfreq = raw.info['sfreq']
+
+    # setup picks of the data
+    picks = pick_types(raw.info, meg=True, eeg=True,
+                       stim=True, ecog=True, seeg=True,
+                       exclude='bads')
+
+    # only sample first 10 seconds, or whole dataset
+    tmin = 0
+    tmax = int(min(len(raw.times), 10 * sfreq))
+    data = raw.get_data(start=tmin, stop=tmax, picks=picks, return_times=False)
+
+    # first bandpass the signal to just between 45 and 65 Hz
+    # only ran if sampling freq is high enough
+    if sfreq > 90:
+        l_freq = 45
+        h_freq = min(sfreq // 2 - 1, 65)
+        data = filter_data(data, sfreq=sfreq,
+                           l_freq=l_freq, h_freq=h_freq)
+
+    # run a multi-taper FFT between Power Line Frequencies of interest
+    raw = mne.io.RawArray(data, info=mne.pick_info(raw.info, sel=picks))
+    usa_psds, usa_freqs = mne.time_frequency.psd_welch(raw,
+                                                       fmin=58, fmax=62,
+                                                       n_jobs=1)
```

Comment:
why not use [psd_array_welch](https://mne.tools/stable/generated/mne.time_frequency.psd_array_welch.html#mne.time_frequency.psd_array_welch) to avoid making the `rawArray` object?